source/common/config/delta_subscription_state.cc

```diff
@@ -214,26 +243,44 @@ DeltaSubscriptionState::getNextRequestWithAck(const UpdateAck& ack) {
 }
 
 void DeltaSubscriptionState::addResourceState(
-    const envoy::service::discovery::v3::Resource& resource) {
+    const envoy::service::discovery::v3::Resource& resource, ResourceType type) {
```

Comment:
This is only called in one place (here: https://github.com/envoyproxy/envoy/blob/f6ee71a5aa413b614198b9966379bcb83d0afea2/source/common/config/delta_subscription_state.cc#L154) , and always with a ResourceType of `ResourceType::ReceivedFromServer`. This effectively makes the call to `effectiveResourceType` a noop; only the `else` branch is needed.

Reply:
I simplified it.